src/TLE.ts

```diff
@@ -874,65 +874,95 @@ export class Parser {
         assert(1 <= stringValue.length, "TLE strings must be at least 1 character.");
         assert(Utilities.isQuoteCharacter(stringValue[0]), "The first character in a TLE string must be a quote character.");
 
-        let leftSquareBracketToken: Token = null;
+        const leftSquareBracketToken: Token = null;
         let expression: Value = null;
-        let rightSquareBracketToken: Token = null;
-        let errors: language.Issue[] = [];
+        const rightSquareBracketToken: Token = null;
+        const errors: language.Issue[] = [];
+        let result: { stringValue: string; leftSquareBracketToken: Token; expression: Value; rightSquareBracketToken: Token; errors: language.Issue[] } = { stringValue, leftSquareBracketToken, expression, rightSquareBracketToken, errors };
 
         if (3 <= stringValue.length && stringValue.substr(1, 2) === "[[") {
```

Comment:
should be able to get rid of the "3 <= stringValue.length" and other string length checks